core/src/main/java/fr/sncf/osrd/api/pathfinding/PathfindingRoutesEndpoint.java

```diff
@@ -102,16 +106,43 @@ public static Pathfinding.Result<SignalingRoute> runPathfinding(
             remainingDistanceEstimator = new RemainingDistanceEstimator(waypoints.get(1));
         }
 
+        Pathfinding.Result<SignalingRoute> pathfinding = computePaths(infra, waypoints, loadingGaugeConstraints,
+                electrificationConstraints, remainingDistanceEstimator);
         // Compute the paths from the entry waypoint to the exit waypoint
-        return new Pathfinding<>(new GraphAdapter<>(infra.getSignalingRouteGraph()))
+        return pathfinding;
+    }
+
+    private static Pathfinding.Result<SignalingRoute> computePaths(SignalingInfra infra,
+             ArrayList<Collection<Pathfinding.EdgeLocation<SignalingRoute>>> waypoints,
+             LoadingGaugeConstraints loadingGaugeConstraints,
+             ElectrificationConstraints electrificationConstraints,
```

Comment:
It would be nice if we don't have to list all the constraints here, eventually there will be many constraints. A set would be nice